x/gamm/pool-models/stableswap/pool.go

```diff
@@ -132,6 +133,23 @@ func (p Pool) scaledPoolReserves() ([]sdk.DecCoin, error) {
 	return scaledReserves, nil
 }
 
+// scaledInput returns scaled input tokens for usage in AMM equations
+func (p Pool) scaledInput(input sdk.Coins) ([]sdk.DecCoin, error) {
```

Comment:
```suggestion
func (p Pool) scaledInput(input sdk.Coin) (sdk.DecCoin, error) {
```

Reply:
I originally had this as `sdk.Coins` to be able to accommodate inputs for all-asset joins as well – will leave it as `sdk.Coin` for now as it's simpler and aligned with the descaling function (which deals with 1 asset) and handle all asset joins at a higher level of abstraction